pkg/apis/core/v1beta1/constants/types_constants.go

```diff
@@ -272,6 +275,21 @@ const (
 	// Note that this annotation is alpha and can be removed anytime without further notice. Only use it if you know
 	// what you do.
 	ShootAlphaControlPlaneScaleDownDisabled = "alpha.control-plane.scaling.shoot.gardener.cloud/scale-down-disabled"
+	// ShootAlphaControlPlaneHighAvailability is a constant for an annotation on the Shoot resource stating that the
+	// high availability setup for the control plane should be enabled.
+	// Note that this annotation is alpha and can be removed anytime without further notice. Only use it if you know
+	// what you do.
+	ShootAlphaControlPlaneHighAvailability = "alpha.control-plane.shoot.gardener.cloud/high-availability"
+	// ShootAlphaControlPlaneHighAvailabilitySingleZone is a specific value that can be set for the shoot control
+	// plane high availability annotation, that allows gardener to spread the shoot control plane across
+	// multiple nodes within a single availability zone if it is possible.
+	ShootAlphaControlPlaneHighAvailabilitySingleZone = "single-zone"
```

Comment:
This should be taken in conjunction with HAControlPlanes feature gate. `single-zone` means that HA is required within a single zone (by having it distributed one per node across nodes), where as `mult-zone` means that HA is required across zones in a multi-zonal setup

Reply:
A single zone HA option is required if the targeted infrastructure essentially has less then 3 zones. Then it's better to just distribute within the same zone than using 2 zones (more details in the upcoming GEP). Why does one want to have a single-zone HA setting at all? Well, besides of the higher fault tolerance, mainly because of zero downtime maintenance.

Reply:
> Well, besides of the higher fault tolerance, mainly because of zero downtime maintenance.

OK, this sounds reasonable, thanks! Maybe the doc strings can be enhanced with such information :)